sensor/admission-control/alerts/sender_test.go

```diff
@@ -0,0 +1,158 @@
+package alerts
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v3"
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+)
+
+func TestAlertSender(t *testing.T) {
+	suite.Run(t, new(alertSenderSuite))
+}
+
+type alertSenderSuite struct {
+	suite.Suite
+	service *alertSenderImpl
+	cl      *fakeAdmissionControlManagementServiceClient
+}
+
+func (s *alertSenderSuite) TestSendAlertsToSensor() {
+	ctx := context.Background()
+	ch := make(chan []*storage.Alert)
+	err := errors.New("error")
+	alerts := createAlertsMessage(2)
+	cases := map[string]struct {
+		numAlertsPerMessage  int
+		responseRequestPairs []*responseRequestPair
+	}{
+		"Stage alerts on error and retry": {
+			responseRequestPairs: []*responseRequestPair{
+				{
+					request:  createAlertsRequest(alerts),
+					response: nil,
+					err:      err,
+				},
+				{
+					request:  createAlertsRequest(alerts),
+					response: &types.Empty{},
+					err:      nil,
+				},
+			},
+		},
+		"Send alerts no error": {
+			responseRequestPairs: []*responseRequestPair{
+				{
+					request:  createAlertsRequest(alerts),
+					response: &types.Empty{},
+					err:      nil,
+				},
+			},
+		},
+	}
+	for testName, c := range cases {
+		s.Run(testName, func() {
+			s.createFakeAlertSender(ch)
+			ctxWithCancel, cancelFunc := context.WithCancel(ctx)
+			wg := &sync.WaitGroup{}
+			s.cl.configureFakeClient(c.responseRequestPairs, wg)
+
+			go func() {
```

Comment:
[nit] rather than starting a goroutine in the test, wouldn't the following yield the same test results?

```go
s.service.Start(ctxWithCancel)
ch <- alerts // This will be read by `s.service` right?
wg.Wait()
// ...
```

If it doesn't work, then maybe the following (untested):
```go
s.service.Start(ctxWithCancel)
select {
case <-ctxWithCancel.Done():
        s.FailNow("...")
case ch <- alerts:
        break
}
wg.Wait()
```